hipscat/margin_utils.py

```diff
@@ -192,3 +196,102 @@ def get_margin(kk, pix, dk):
     nb = list(get_edge(dk, px, edge) for edge, px in zip(which, n) if px != -1)
     nb = np.concatenate(nb)
     return nb
+
+def get_margin_scale(k, margin_threshold):
+    resolution = hp.nside2resol(2**k, arcmin=True) / 60.
+    resolution_and_thresh = resolution + (margin_threshold)
+    margin_area = resolution_and_thresh ** 2
+    pixel_area = hp.pixelfunc.nside2pixarea(2**k, degrees=True)
+    scale = margin_area / pixel_area
+    return scale
+
+def get_margin_bounds_and_wcs(k, pix, scale, step=10):
+    
+    pixel_boundaries = hp.vec2dir(hp.boundaries(2**k, pix, step=step, nest=True), lonlat=True)
+
+    n_samples = len(pixel_boundaries[0])
+    centroid_lon = np.sum(pixel_boundaries[0]) / n_samples
+    centroid_lat = np.sum(pixel_boundaries[1]) / n_samples
+    translate_lon = centroid_lon - (centroid_lon * scale)
```

Comment:
`scale` is a proportion of area, but is being used here to adjust a length. I think? So you might actually want `root(scale)` to apply in 1D.

Say in cartesian coordinates, you've got a 4x4 grid, and you want to have a larger bounding box of 6x6. The `scale` would come out to `36/16 = 2.25`, but when adjusting the x or y, you'd want to use a scaling factor of `6/4 = 1.5`.

Reply:
it's not adjusting a length here, it's applying the scale factor to the center point of the healpixel. *technically* this step is applying an affine scale matrix to the point in homogenous coordinate space, like
```
[scale, 0, 0]    [centroid_lon]
[0, scale, 0] x  [centroid_lat]
[0,     0, 1]    [           1]
```
then finding the differences between this new transformed coordinate and the original centroid position.

however since the only operation that we care about in the affine transform multipling the centroid coordinates against the scale factor and then finding the difference between that and the original, we can just do it in one line instead of writing out the full matrix multiplication.